### PR TITLE
[SIMD.js] Fix offset of store of typed array by adding tarray.byteOffset...

### PR DESCRIPTION
**Pull request by chuan9**
....

Add test case for byteOffset of float32x4Array.

BUG=XWALK-3155

### CODE REVIEW

File: test/mjsunit/simd/float32x4.js [hunk 1]
```diff
@@ -943,6 +943,7 @@ function testArrayOfFloat32x4() {
 
 testArrayOfFloat32x4();
 
+<<<<<<< HEAD
```
Comment (rakuco) [inline test/mjsunit/simd/float32x4.js]: Wrong.

File: test/mjsunit/simd/float32x4.js [hunk 2]
```diff
@@ -953,6 +954,8 @@ function testFloat32x4ArrayIndex() {
 
 testFloat32x4ArrayIndex();
 
+=======
```
Comment (rakuco) [inline test/mjsunit/simd/float32x4.js]: And wrong. Please make sure you've resolved the conflict correctly.

**Comment by huningxin** [comment]
LGTM.

**Comment by huningxin** [comment]
I am going to merge it. But as the master is updated, please rebase your patch. Thanks.

**Pull request merged**